maven/src/org/netbeans/modules/maven/classpath/AbstractProjectClassPathImpl.java

```diff
@@ -206,4 +211,38 @@ public void removePropertyChangeListener(PropertyChangeListener propertyChangeLi
         return project.hashCode() ^ getClass().hashCode();
     }
 
+    /**
+     * Like {@link Artifact#getFile} but when a timestamped snapshot is locally downloaded, uses that instead.
+     */
+    protected static File getFile(Artifact art) {
+        File f = art.getFile();
```

Comment:
Am I basically OK with the overall change, but if I am the one who merges it, then I'd like to see a test for this new method:
- subclass `NbTestCase`
- in `setUp` use `getWorkDir()` and create illustrative file structure 
- in `testXXX` method(s) verify the `getFile` works as expected.
Once such test is created, ping me and I merge. Alternatively find another person to merge your change.

Reply:
Ack. Have not had a moment to work on this yet.